File: payment_service.py
from flask import Flask, request, jsonify
from flask_cors import CORS
import uuid
import random

app = Flask(__name__)
CORS(app)

payments = {}

@app.route('/payments', methods=['POST'])
def process_payment():
    data = request.json
    required = ['booking_id', 'amount', 'payment_method']
    if not all(field in data for field in required):
        return jsonify({"error": "Missing fields"}), 400
    
    success = random.random() < 0.95
    payment_id = str(uuid.uuid4())[:8]
    
    payment = {
        "payment_id": payment_id,
        "booking_id": data['booking_id'],
        "amount": data['amount'],
        "status": "SUCCESS" if success else "FAILED",
        "payment_method": data['payment_method']
    }
    payments[payment_id] = payment
    
    return jsonify(payment), 200 if success else 402

@app.route('/payments/<payment_id>', methods=['GET'])
def get_payment(payment_id):
    payment = payments.get(payment_id)
    if not payment:
        return jsonify({"error": "Payment not found"}), 404
    return jsonify(payment)

@app.route('/notify', methods=['POST'])
def send_notification():
    data = request.json
    print(f"[NOTIFICATION] To: {data.get('email')} | Subject: {data.get('subject')}")
    return jsonify({"status": "sent"}), 200

if __name__ == '__main__':
    app.run(host='0.0.0.0', port=5003, debug=True)

File: payment_service.js
const express = require('express');
const cors = require('cors');
const { v4: uuidv4 } = require('uuid');

const app = express();
app.use(cors());
app.use(express.json());

const payments = {};

app.post('/payments', (req, res) => {
    const { booking_id, amount, payment_method } = req.body;
    
    if (!booking_id || !amount || !payment_method) {
        return res.status(400).json({ error: "Missing fields" });
    }
    
    const success = Math.random() < 0.95;
    const payment_id = uuidv4().slice(0, 8);
    
    const payment = {
        payment_id,
        booking_id,
        amount,
        status: success ? "SUCCESS" : "FAILED",
        payment_method
    };
    payments[payment_id] = payment;
    
    res.status(success ? 200 : 402).json(payment);
});

app.listen(5003, () => {
    console.log('Payment service running on port 5003');
});
